### PR TITLE
✨ Include amount of people recipe is intended for

### PR DESCRIPTION
**Pull request by lisamittmann**
- extend data model to include additional attribute
- add new placeholder in frontend component
- make controller path naming more consistent with including `api` at beginning
- update service test to include recipe object with all fields set

### CODE REVIEW

File: backend/src/main/java/de/neuefische/teddyscoronadiaries/model/recipe/Recipe.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
     private String name;
     private String imageUrl;
     private String diaryEntry;
+    private String amountOfPeople;
```
Comment (jamarob) [inline backend/src/main/java/de/neuefische/teddyscoronadiaries/model/recipe/Recipe.java]: Maybe `numberOfPeople`? Amount sounds strange to me 😄

File: backend/src/test/java/de/neuefische/teddyscoronadiaries/controller/RecipeControllerTest.java [hunk 2]
```diff
@@ -52,6 +52,7 @@ public void getRecipeShouldReturnRecipe(){
                 .imageUrl("some-image-url")
                 .diaryEntry("wenn absolut nichts mehr geht, dann ist es Zeit für Schokokuchen")
                 .quarantineDay(25)
+                .amountOfPeople("Für 20cm SPringform")
```
Comment (jamarob) [inline backend/src/test/java/de/neuefische/teddyscoronadiaries/controller/RecipeControllerTest.java]:
```suggestion
                .amountOfPeople("Für 20cm Springform")
```
Hm.. if this is more general, maybe call it just `amount`?

File: backend/src/test/java/de/neuefische/teddyscoronadiaries/controller/RecipeControllerTest.java [hunk 3]
```diff
@@ -71,6 +72,7 @@ public void getRecipeShouldReturnRecipe(){
                 .name("Himmlischer Schokokuchen")
                 .imageUrl("some-image-url")
                 .diaryEntry("wenn absolut nichts mehr geht, dann ist es Zeit für Schokokuchen")
+                .amountOfPeople("Für 20cm SPringform")
```
Comment (jamarob) [inline backend/src/test/java/de/neuefische/teddyscoronadiaries/controller/RecipeControllerTest.java]:
```suggestion
                .amountOfPeople("Für 20cm Springform")
```

File: backend/src/test/java/de/neuefische/teddyscoronadiaries/service/RecipeServiceTest.java [hunk 4]
```diff
@@ -29,6 +32,15 @@ public void getRecipeShouldReturnRecipe(){
                         .name("Himmlischer Schokokuchen")
                         .imageUrl("some-image-url")
                         .diaryEntry("wenn absolut nichts mehr geht, dann ist es Zeit für Schokokuchen")
+                        .quarantineDay(25)
+                        .amountOfPeople("Für 20cm SPringform")
```
Comment (jamarob) [inline backend/src/test/java/de/neuefische/teddyscoronadiaries/service/RecipeServiceTest.java]:
```suggestion
                        .amountOfPeople("Für 20cm Springform")
```

File: backend/src/test/java/de/neuefische/teddyscoronadiaries/service/RecipeServiceTest.java [hunk 5]
```diff
@@ -41,6 +53,15 @@ public void getRecipeShouldReturnRecipe(){
                 .name("Himmlischer Schokokuchen")
                 .imageUrl("some-image-url")
                 .diaryEntry("wenn absolut nichts mehr geht, dann ist es Zeit für Schokokuchen")
+                .quarantineDay(25)
+                .amountOfPeople("Für 20cm SPringform")
```
Comment (jamarob) [inline backend/src/test/java/de/neuefische/teddyscoronadiaries/service/RecipeServiceTest.java]:
```suggestion
                .amountOfPeople("Für 20cm Springform")
```

File: frontend/src/components/RecipePreparation.js [hunk 6]
```diff
@@ -1,11 +1,11 @@
 import styled from 'styled-components/macro'
 
-export default function RecipePreparation({ingredients, preparationSteps}) {
+export default function RecipePreparation({amountOfPeople, ingredients, preparationSteps}) {
     return (
         <PreparationGrid>
             <div>
                 <ul>
-                    <p>Zutaten</p>
+                    <p>{amountOfPeople}</p>
```
Comment (jamarob) [inline frontend/src/components/RecipePreparation.js]: Only `li` are allowed as direct children of `ol` and `ul`! The browser will render it fine, but it breaks html semantics.

File: frontend/src/pages/RecipePage.js [hunk 7]
```diff
@@ -22,7 +22,7 @@ export default function RecipePage() {
                 <img src={recipe.imageUrl} alt={recipe.name}/>
                 <RecipeHeader recipeName={recipe.name} recipeDescription={recipe.diaryEntry}/>
                 <RecipeCoronaDetails quarantineDay={recipe.quarantineDay}/>
-                <RecipePreparation ingredients={recipe.ingredients} preparationSteps={recipe.steps}/>
+                <RecipePreparation ingredients={recipe.ingredients} preparationSteps={recipe.steps} amountOfPeople={recipe.amountOfPeople}/>
```
Comment (jamarob) [inline frontend/src/pages/RecipePage.js]: You could simplify this by renaming the `preparationSteps` prop to just `steps` and then do
```suggestion
                <RecipePreparation {...recipe}/>
```
Spread operator is your friend when object keys and prop names match ;)

**Review by jamarob — CHANGES_REQUESTED** [review]
Please change the list to follow html semantics, otherwise looks very good, only nit picky comments from me 🐿️

File: backend/src/test/java/de/neuefische/teddyscoronadiaries/controller/RecipeControllerTest.java (hunk 2 above)
Comment (lisamittmann) [reply backend/src/test/java/de/neuefische/teddyscoronadiaries/controller/RecipeControllerTest.java]: how about `intendedFor` ?

File: frontend/src/pages/RecipePage.js (hunk 7 above)
Comment (lisamittmann) [reply frontend/src/pages/RecipePage.js]: yes great point, thx for pointing that out 👍🏻

File: frontend/src/components/RecipePreparation.js (hunk 6 above)
Comment (lisamittmann) [reply frontend/src/components/RecipePreparation.js]: some day I'll go straight to HTML + JS hell 😈 🤣
fixed that

File: frontend/src/components/RecipeSteps.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import styled from 'styled-components/macro'
+
+export default function RecipeSteps({steps}) {
+    return (
+        <StepsWrapper>
+            <p>Zubereitung</p>
+            <ol>
+                {steps.sort((oneStep, anotherStep) => oneStep.stepNumber.localeCompare(anotherStep.stepNumber))
```
Comment (jamarob) [inline frontend/src/components/RecipeSteps.js]: Careful, `sort` will modify the array in place, maybe make a quick copy with `.slice()` before, as we don't want to change props (`steps`) directly.

**Review by jamarob — CHANGES_REQUESTED** [review]
Better not modify input arguments (props), apart from that: good job!

**Review by jamarob — APPROVED** [review]
🍜 🌟

**Pull request merged**